packages/cache/src/cache.ts

```diff
@@ -181,18 +171,45 @@ export async function saveCache(
     if (core.isDebug()) {
       await listTar(archivePath, compressionMethod)
     }
-
     const fileSizeLimit = 10 * 1024 * 1024 * 1024 // 10GB per repo limit
     const archiveFileSize = utils.getArchiveFileSizeInBytes(archivePath)
     core.debug(`File Size: ${archiveFileSize}`)
-    if (archiveFileSize > fileSizeLimit) {
+    // For GHES, this check will take place in ReserveCache API with enterprise file size limit
+    if (archiveFileSize > fileSizeLimit && !utils.isGhes()) {
```

Comment:
should we check the size on GHES and throw an appropriate error here? May be compromising the GHES experience here if they don't change the cache size. What happens if we try to upload a file that is too big? Does it try to upload and fail.

Reply:
As we dont have the limit for GHES repos here, we moved the check to actions-dotnet. If the file is too big we will thrown an error which is being handled as 400 here.

[ADR](https://github.com/github/c2c-actions/blob/c5e9d502a0fb5c1f806366347c8ad10f695e8816/docs/teams/platform/adrs/artifact-cache/0704-set-artifact-cache-size-limit.md)
[PR link to this change. ](https://github.com/github/actions-dotnet/pull/11557)